presto-orc/src/main/java/com/facebook/presto/orc/reader/StructStreamReader.java

```diff
@@ -96,18 +97,28 @@ public Block readBlock(Type type)
         Block[] blocks = new Block[typeParameters.size()];
         if (presentStream == null) {
             for (int i = 0; i < typeParameters.size(); i++) {
-                StreamReader structField = structFields[i];
-                structField.prepareNextRead(nextBatchSize);
-                blocks[i] = structField.readBlock(typeParameters.get(i));
+                if (i < structFields.length) {
+                    StreamReader structField = structFields[i];
+                    structField.prepareNextRead(nextBatchSize);
+                    blocks[i] = structField.readBlock(typeParameters.get(i));
+                }
+                else {
+                    blocks[i] = getNullBlock(typeParameters.get(i), blocks[0].getPositionCount());
```

Comment:
This assumes `structFields.length >= 1`. Is there a way to avoid  this assumption?